distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1618,6 +1658,172 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
                 .resolve(CSW_SOURCE_ID + "-" + metacardId + ".ser")), is(false));
     }
 
+    @Test
+    public void testTwoUsersSameProductRetrySuccess() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+        CometDClient cometDClient2 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+
+        setupCswServerForSuccess(cometDClient1, filename);
+        setupCswServerForSuccess(cometDClient2, filename);
+
+        // where 3 is the number of expectations and 9 is the number of activities
+        checkExpectations(cometDClient1, 3, 9);
+        checkExpectations(cometDClient2, 3, 9);
+
+        List<String> activities = cometDClient1.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        boolean activityFound = foundExpectedActivity(activities, filename, "COMPLETE");
+        assertThat(activityFound, is(equalTo(true)));
+
+        activities = cometDClient2.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        activityFound = foundExpectedActivity(activities, filename, "COMPLETE");
+        assertThat(activityFound, is(equalTo(true)));
+
+        List<String> notifications = cometDClient1.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        boolean notificationFound = foundExpectedActivity(notifications, filename, "completed");
+        assertThat(notificationFound, is(equalTo(true)));
+
+        notifications = cometDClient2.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        notificationFound = foundExpectedActivity(notifications, filename, "completed");
+        assertThat(notificationFound, is(equalTo(true)));
+
+    }
+
+    @Test
+    public void testTwoUsersSameProductRetryFailure() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+        CometDClient cometDClient2 = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL));
+
+        setupCswServerForFailure(cometDClient1, filename);
+        setupCswServerForFailure(cometDClient2, filename);
+
+        // where 4 is the number of expectations and 8 is the number of activities
+        checkExpectations(cometDClient1, 4, 8);
+        checkExpectations(cometDClient2, 4, 8);
```

Comment:
@bantillo also here - need to know _why_ we picked these constants (if I'm understanding the above critique correctly @coyotesqrl and @shaundmorris)